### PR TITLE
api: return ErrXXX.With instead of fmt.Errorf

### PR DESCRIPTION
**Pull request by altergui**
a couple of fmt.Errorf were introduced recently, fix them. (they don't produce JSON replies, breaking interop)

### CODE REVIEW

File: api/censuses.go [hunk 1]
```diff
@@ -736,7 +736,7 @@ func (a *API) censusVerifyHandler(msg *apirest.APIdata, ctx *httprouter.HTTPCont
 
 	valid, err := ref.Tree().VerifyProof(leafKey, cdata.Value, cdata.Proof, cdata.Root)
 	if err != nil {
-		return fmt.Errorf("something was wrong during census proof verification: %w", err)
+		return ErrCensusProofVerificationFailed.WithErr(err)
```
Comment (altergui) [inline api/censuses.go:739]: before:
```
> GET /v2/censuses/dd68b419fda0d568f439c35f11fe11dc4f2ea0858d3a451d5008c5947377b91d/proof/38d2BC91B89928f78cBaB3e4b1949e28787eC7a3 HTTP/1.1
> Host: 127.0.0.1:9090
> User-Agent: curl/7.74.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Internal Server Error
< Content-Length: 112
< Content-Type: application/json
< Vary: Origin
< Date: Thu, 23 Mar 2023 15:19:54 GMT
< 
something was wrong during census proof generation: key does not exist ed947252f130d1d34f2391f7098383d9c6297638
```

after:
```
> GET /v2/censuses/dd68b419fda0d568f439c35f11fe11dc4f2ea0858d3a451d5008c5947377b91d/proof/38d2BC91B89928f78cBaB3e4b1949e28787eC7a3 HTTP/1.1
> Host: 127.0.0.1:9090
> User-Agent: curl/7.74.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Content-Length: 114
< Content-Type: application/json
< Vary: Origin
< Date: Thu, 23 Mar 2023 15:26:40 GMT
< 
{"error":"key not found in census: ed947252f130d1d34f2391f7098383d9c6297638b414fb648544612edf4b006f","code":4048}
```

File: api/censuses.go [hunk 2]
```diff
@@ -661,7 +661,7 @@ func (a *API) censusProofHandler(msg *apirest.APIdata, ctx *httprouter.HTTPConte
 	}
 	leafV, siblings, err := ref.Tree().GenProof(leafKey)
 	if err != nil {
-		return fmt.Errorf("something was wrong during census proof generation: %w", err)
+		return ErrKeyNotFoundInCensus.With(hex.EncodeToString(leafKey))
```
Comment (altergui) [inline api/censuses.go:664]: before:
```
> GET /v2/censuses/dd68b419fda0d568f439c35f11fe11dc4f2ea0858d3a451d5008c5947377b91d/proof/38d2BC91B89928f78cBaB3e4b1949e28787eC7a3 HTTP/1.1
> Host: 127.0.0.1:9090
> User-Agent: curl/7.74.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Internal Server Error
< Content-Length: 112
< Content-Type: application/json
< Vary: Origin
< Date: Thu, 23 Mar 2023 15:19:54 GMT
< 
something was wrong during census proof generation: key does not exist ed947252f130d1d34f2391f7098383d9c6297638
```

after:
```
> GET /v2/censuses/dd68b419fda0d568f439c35f11fe11dc4f2ea0858d3a451d5008c5947377b91d/proof/38d2BC91B89928f78cBaB3e4b1949e28787eC7a3 HTTP/1.1
> Host: 127.0.0.1:9090
> User-Agent: curl/7.74.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Content-Length: 114
< Content-Type: application/json
< Vary: Origin
< Date: Thu, 23 Mar 2023 15:26:40 GMT
< 
{"error":"key not found in census: ed947252f130d1d34f2391f7098383d9c6297638b414fb648544612edf4b006f","code":4048}
```

**Comment by p4u** [comment]
Great! @lucasmenendez git blame points to you ;) So all API errors from now on should be handled with the new Error interface.

**Review by p4u — APPROVED** [review]

**Comment by p4u** [comment]
@altergui can you fix the conflicts and merge this PR?

**Comment by altergui** [comment]
i fixed the conflict locally but i can't push the branch, github is down. i'll merge this ASAP

**Pull request merged**